docs/getting-started/fundamentals-block-development/registration-of-a-block.md

```diff
@@ -0,0 +1,93 @@
+# Registration of a block
+
+A block is usually registered through a plugin in both the server and the client using its `block.json` metadata. 
+
+Although technically, blocks could be registered only on the client, **registering blocks in both the server and the client is a general recommendation**, as some features like Dynamic Rendering, Block Hooks, or Block style variations won't work properly without a server registration of the block. 
```

Comment:
I haven't reviewed everything, but the existing text sounds about right.

I don't know how deep you want to go into this, or if you rather link to existing documents. What I could add is that "global styles" (styles generated from `theme.json`) require blocks to be registered in the server – otherwise, it's like they don't exist, and any styles assigned to them in `theme.json` will be ignored. This only matters if the block aims to be styled via `theme.json`, of course.

I presume the same would be true for all features that generate styles in the server (block supports, layout, style engine, etc.). @aaronrobertshaw @ramonjd @andrewserong @tellthemachines have implemented most of it. They know more than I do on this topic!

Reply:
Excellent feedback Andre. Much appreciate inviting other implementers to share more context 🙇🏻

Reply:
Thanks @gziolo @oandregal for the feedback! 
I've updated the PR taking this feedback into account. 
Let me know your thoughts.

> I don't know how deep you want to go into this, or if you rather link to existing documents.

The purpose of this "Getting Started > Fundamentals of Block Development" is to introduce key ideas to help developers have a better "onboarding" experience for block development. Links to other parts of the Handbook will be used for more details, but in some specific cases, the content will be "moved" to this section.

Reply:
Hi folks, thanks for the ping, and great work on the docs. 

In relation to the style engine, along with the general [README/usage](https://github.com/WordPress/gutenberg/blob/trunk/packages/style-engine/README.md) docs there is some block-supports-specific documentation written here:

https://github.com/WordPress/gutenberg/blob/trunk/packages/style-engine/docs/using-the-style-engine-with-block-supports.md

And the handbook entry for [block supports](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/).

As @juanmaguitar suggests, maybe it's enough to refer and link to the relevant docs as appropriate should folks wish to enable block supports features to their newly-registered block?

P.S., Going over this stuff made me realize I need to update the style engine block supports docs so I have a [PR](https://github.com/WordPress/gutenberg/pull/56528) to do that 😆 So thank you!